src/cascadia/TerminalSettingsModel/JsonUtils.h

```diff
@@ -259,6 +473,59 @@ namespace Microsoft::Terminal::Settings::Model::JsonUtils
         }
     };
 
+    template<typename T>
+    struct ConversionTrait<winrt::Windows::Foundation::Collections::IVector<T>>
+    {
+        winrt::Windows::Foundation::Collections::IVector<T> FromJson(const Json::Value& json)
+        {
+            winrt::Windows::Foundation::Collections::IVector<T> vec = winrt::single_threaded_vector<T>();
+
+            for (auto it = json.begin(), end = json.end(); it != end; ++it)
+            {
+                vec.Append(GetValue<T>(*it));
+            }
+
+            return vec;

```

Comment:
```suggestion
            ConversionTrait<std::vector<T>> trait;
            return winrt::single_threaded_vector(trait.FromJson(json));
```

Reply:
Does the `single_threaded_vector` do a full copy of the input? I figured I shouldn't make an intermediate.

Reply:
It's a move-only constructor actually. For instance this - which would create a copy - doesn't compile:
```cpp
std::vector<T> vec;
return winrt::single_threaded_vector(vec);
```

You actually have to write:
```cpp
std::vector<T> vec;
return winrt::single_threaded_vector(std::move(vec));
```
...in order to explicitly prevent a copy.